app/src/main/java/com/timchenko/playlistmaker/ui/media/PlaylistDetailsFragment.kt

```diff
@@ -0,0 +1,278 @@
+    private lateinit var playlistTrackAdapter: PlaylistTrackAdapter
+    private lateinit var playlistTrackAdapterListener: PlaylistTrackAdapter.Listener
+
+    private lateinit var bottomSheetMenu: BottomSheetBehavior<LinearLayout>
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentPlaylistDetailsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val playlistID = requireArguments().getInt(PLAYLIST_ID, 0)
+        if (playlistID > 0) {
+            viewModel.getPlaylist(playlistID)
+            viewModel.getTracksForPlaylist(playlistID)
+        }
+        else {
+            findNavController().popBackStack()
+        }
+        viewModel.observePlaylist().observe(viewLifecycleOwner) {
+            initPlaylist(it)
+        }
+        viewModel.observeTracks().observe(viewLifecycleOwner) {
+            initPlaylistTracks(it)
+        }
+        playlistTrackAdapterListener = object : PlaylistTrackAdapter.Listener {
+            override fun onShortClick(track: Track) {
+                if (clickDebounce()) {
+                    val displayIntent = Intent(requireContext(), AudioPlayerActivity::class.java)
+                    displayIntent.putExtra("track", track)
+                    startActivity(displayIntent)
+                }
+            }
+            override fun onLongClick(trackId: Int): Boolean {
+                showDeleteTrackDialog(trackId)
+                return true
+            }
+
+        }
+        playlistTrackAdapter = PlaylistTrackAdapter(playlistTrackAdapterListener)
+        binding.recyclerPlaylistTracks.adapter = playlistTrackAdapter
+
+        bottomSheetMenu = BottomSheetBehavior.from(binding.BottomSheetPlaylistMenu).apply {
+            state = BottomSheetBehavior.STATE_COLLAPSED
+        }
+    }
+
+    private fun initPlaylist(playlist: Playlist) {
+        binding.playlistName.text = playlist.name
+        binding.playlistNameMenu.text = playlist.name
+        binding.playlistDescription.text = playlist.description
+        binding.playlistDescriptionMenu.text = playlist.description
+        numberOfTracksString = playlist.tracksCounter.let {
+            resources.getQuantityString(
+                R.plurals.track_counter,
+                it,
+                playlist.tracksCounter
+            )
+        }
+        val numberOfMinutesString: String = resources.getQuantityString(
+            R.plurals.minutes_counter,
+            Formatter.convertMillisToMinutes(playlist.trackTimerMillis).toInt(),
+            Formatter.convertMillisToMinutes(playlist.trackTimerMillis).toInt()
+        )
+
+        binding.playlistStats.text = numberOfMinutesString  + " \u2022 " + numberOfTracksString
```

Comment:
🍏 В памяти будут создаваться промежуточные строки из-за использования оператора "+".  Можно использовать StringBuilder, который сформирует только результирующую строку. В kotlin есть удобный метод buildString, который под капотом использует StringBuilder. https://habr.com/ru/articles/260767/ https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html
Либо можно использовать Kotlin templates:
```suggestion
       binding.playlistStats.text = "$numberOfMinutesString \u2022 $numberOfTracksString"
```